### ISSUE TITLE
UselessPostfixExpression false positives

### ISSUE THREAD

**Issue by olivierlemasle** [arturbosch/detekt#857]
`UselessPostfixExpression` reports issues in the following cases:

Case 1:
```kotlin
fun testProperty(): Int {
   return shouldNotBeNull!!.field
}
``` 

Case 2:
```kotlin
class Foo(var i: Int = 0) {
   fun getIdAndIncrement(): Int {
      return i++
   }
}
```
## Expected Behavior
`UselessPostfixExpression` should not report any issue in both cases (case 1: this is not operator ++ or --; case 2: the incrementation of `i` is not useless because it has a side-effect, incrementing class field `i`)

## Current Behavior
In both cases, an issue `UselessPostfixExpression` is reported.

For case 1, similar issue  #491 was fixed, but it does not fix the case where the `expression!!` is part of another expression.

For case 2, similar issue #466 was fixed, but it does not fix the case where the incremented class field is defined in the primary constructor.

## Your Environment
* Version used: 1.0.0.RC6-4
* Operating System and version: GNU/Linux, Fedora 28
